### PR TITLE
fix waitServer test by using in-memory connections. Closes #118

### LINKED ISSUE #118: Tests failing in CI from time to time
We already had some issues with this in the past. Those are tests where the gRPC wait server is involved.

Here is an example of how it looks like:

https://github.com/metal-stack/metal-api/pull/115/checks?check_run_id=1242007616

### PR DESCRIPTION
**Pull request by kolsa**

### CODE REVIEW

**Comment by majst01** [comment]
LGTM can we merge this ?

**Review by Gerrit91 — APPROVED** [review]

**Comment by Gerrit91** [comment]
I think yes!

File: cmd/metal-api/internal/grpc/waitServer.go [hunk 1]
```diff
@@ -247,7 +247,9 @@ func (s *WaitServer) Wait(req *v1.WaitRequest, srv v1.Wait_WaitServer) error {
 	defer func() {
 		s.queueLock.Lock()
 		delete(s.queue, machineID)
-		close(can)
+		if err != nil {
```
Comment (majst01) [inline cmd/metal-api/internal/grpc/waitServer.go:250]: cant immediately see where this error is coming from

Comment (kolsa) [reply cmd/metal-api/internal/grpc/waitServer.go:250]: This happened a few times during 100+ subsequent test runs

Comment (kolsa) [reply cmd/metal-api/internal/grpc/waitServer.go:250]: I'll revert this for now and try to test such a scenario

Comment (majst01) [reply cmd/metal-api/internal/grpc/waitServer.go]: OK, now the tests fail, this is weird. so we need the errcheck

**Comment by Gerrit91** [comment]
Probably not required anymore due to #131.

**Pull request closed without merging**